sdks/python/apache_beam/runners/worker/statesampler.pyx

```diff
@@ -150,11 +161,31 @@ cdef class StateSampler(object):
           # without the GIL.
           nsecs_ptr = &(<ScopedState>PyList_GET_ITEM(
               self.scoped_states_by_index, self.current_state_index)).nsecs
+          lull_nsecs_ptr = &(<ScopedState>PyList_GET_ITEM(
+              self.scoped_states_by_index, self.current_state_index)).lull_nsecs
```

Comment:
<!--thread_id:cc_143091727_t; commit:95830a027670626ce0f081991d56dac60c8ee8d1; resolved:0-->
<!--section:context-quote-->
> **pabloem** wrote:
> I'm sorry, I am not sure how we'd use transition count to track this. Can you give me a snippet for it? Or I could, instead, add a `time_since_last_transition` property to `StateSampler`, and have similar code. Let me know what you think.

<!--section:body-->
From what I understand, the point of having the `lull_nsecs` property is to track how much time has been spent in the current stage.  This can be inferred by looking at `state_transition_count` and accumulating time spent as long as the `state_transition_count` value hasn't changed.  This way, we avoid adding the extra property and can still track how much time is spent without a transition.

Reply:
We want to track how much time has been spent *stuck* in the current step, without transitioning out of it, right. I've added a property to the `StateSampler` instead. It could be done with a couple local variables instead - would you prefer that?